implemented.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+)
+
+// implemented checks method already implemetned.
+// ifns could be requested by calling ifuncs().
+func implemented(ifns []Func, fnName string) bool {
+	for _, ifn := range ifns {
+		if fnName == ifn.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// ifuncs returns list of Func which already implemented.
+func ifuncs(fns []Func, typeTitle string, srcDir string) ([]Func, error) {
+
+	fset := token.NewFileSet()
+	pkgs, err := parser.ParseDir(fset, srcDir, nil, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	var ifns []Func
+
+	// getReceiver returns title of struct to which belongs the method
+	getReceiver := func(mf *ast.FuncDecl) string {
+		if mf.Recv != nil {
+			for _, v := range mf.Recv.List {
+				switch xv := v.Type.(type) {
```

Comment:
This might be simpler and safer as two conditionals:

```go
// strip leading *
if _, ok := v.(*ast.StarExpr); ok {
  v = v.X
}
if _, ok := v.(*ast.Ident); ok {
  return v.Name
}
```

Reply:
We can't assign `v.X` to `v`, because `v` is not an `interface{}`.

Reply:
I think josh's point is more that instead of having a switch statement the flow can be simplified by first casting the start expr to ident then treating both cases as ident thus reducing the number of returns to 1
```go
func getName(v interface{}) string {
// strip leading *
  if n, ok := v.Type.(*ast.StarExpr); ok {
    v = n.X
  }
  if ident, ok := v.Type.(*ast.Ident); ok {
    return ident.Name
  }
  return "" // or panic
}
```

Reply:
I understood at first time. But as I thought @josharian meant to do it inside `getReceiver` func and not move this code to external func like @sevlki propose.

If I replace my `switch` to `if` I'll receive compiler error:
`cannot use n.X (variable of type ast.Expr) as * ast.Field value in assignment`

So this is a small piece of code I think It's not needed create new func for it. And propose don't change it.

Reply:
sorry, I didn't mean to backseat drive your PR.  I was just looking at this exact problem today and was delighted to see the PR, thought some more activity might help speed things alone :) just ended up creating noise :frowning_face: